src/java/fr/paris/lutece/plugins/workflow/modules/userassignment/service/task/TaskUserAssignmentNotification.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.servlet.http.HttpServletRequest;
+
+
+import fr.paris.lutece.plugins.workflow.modules.userassignment.business.TaskUserAssignmentNotificationConfig;
+import fr.paris.lutece.plugins.workflowcore.business.resource.ResourceHistory;
+import fr.paris.lutece.plugins.workflowcore.service.config.ITaskConfigService;
+import fr.paris.lutece.plugins.workflowcore.service.resource.IResourceHistoryService;
+import fr.paris.lutece.plugins.workflowcore.service.resource.ResourceHistoryService;
+import fr.paris.lutece.plugins.workflowcore.service.task.ITask;
+import fr.paris.lutece.plugins.workflowcore.service.task.ITaskService;
+import fr.paris.lutece.plugins.workflowcore.service.task.SimpleTask;
+import fr.paris.lutece.plugins.workflowcore.service.task.TaskService;
+import fr.paris.lutece.portal.business.user.AdminUser;
+import fr.paris.lutece.portal.service.admin.AdminUserService;
+import fr.paris.lutece.portal.service.i18n.I18nService;
+import fr.paris.lutece.portal.service.mail.MailService;
+import fr.paris.lutece.portal.service.template.AppTemplateService;
+import fr.paris.lutece.util.html.HtmlTemplate;
+
+/**
+ * TaskUserAssignmentNotification
+ *
+ */
+public class TaskUserAssignmentNotification extends SimpleTask
+{
+    // Message
+    private static final String MESSAGE_TASK_TITLE 	= "module.workflow.userassignment.task_user_assignment_notification.title";
+
+    //MARKS
+    private static final String MARK_RESOURCE_ID 	= "resourceId";
+    private static final String MARK_RESOURCE_TYPE 	= "resourceType";
+    
+	private static final String BEAN_CONFIG 		= "workflow-userassignment.taskUserAssignmentNotificationConfigService";
+	
+    @Inject
+    @Named( BEAN_CONFIG )
+    private ITaskConfigService _taskConfigService;
+    
+    @Inject
+    @Named( ResourceHistoryService.BEAN_SERVICE )
+    private IResourceHistoryService _resourceHistoryService;
+    
+    @Inject
+    @Named ( TaskService.BEAN_SERVICE )
+    private ITaskService _taskService;
+    
+    @Inject
+    private IAssignUserResourceTaskService _userResourceService;
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getTitle( Locale locale )
+    {
+        return I18nService.getLocalizedString( MESSAGE_TASK_TITLE, locale );
+    }
+
+
+	@Override
+	public void processTask( int nIdResourceHistory, HttpServletRequest request, Locale locale )
+	{
+		ResourceHistory resourceHistory = _resourceHistoryService.findByPrimaryKey( nIdResourceHistory );
+		
+		TaskUserAssignmentNotificationConfig config = _taskConfigService.findByPrimaryKey( getId( ) );
+		
+		if ( resourceHistory != null && config != null )
+		{			
+			List<ITask> listTask = _taskService.getListTaskByIdAction( resourceHistory.getAction( ).getId( ), locale );
+			
+			if ( listTask != null )
+			{
+				for ( ITask task : listTask )
+				{
+					if ( task instanceof AssignUserResourceTask )
```

Comment:
We should log if we can't find any instance of AssignUserResourceTask as an error.
This task should also be prior (in display order) to the notification task.

Reply:
Maybe the AssignUserResourceTask task list of the action could be listed when creating the notification task ?